dd-sdk-android-ktx/src/test/kotlin/com/datadog/android/ktx/coroutine/CoroutineExtTest.kt

```diff
@@ -0,0 +1,479 @@
+            withContext(Dispatchers.IO) {
+                lambdaCalled = true
+            }
+        }
+        Thread.sleep(100)
+
+        // Then
+        assertThat(lambdaCalled).isTrue()
+        verify(mockSpanBuilder).asChildOf(mockParentSpan)
+        inOrder(mockSpan, mockScope) {
+            verify(mockSpan).finish()
+            verify(mockScope).close()
+        }
+    }
+
+    @Test
+    fun `M create span around coroutine W runBlockingTraced() without parents`() {
+        // Given
+        var lambdaCalled = false
+        whenever(mockTracer.activeSpan()) doReturn null
+        whenever(mockSpanBuilder.asChildOf(null as Span?)) doReturn mockSpanBuilder
+
+        // When
+        runBlockingTraced(fakeOperationName, Dispatchers.Default) {
+            withContext(Dispatchers.IO) {
+                lambdaCalled = true
+            }
+        }
+        Thread.sleep(100)
+
+        // Then
+        assertThat(lambdaCalled).isTrue()
+        verify(mockSpanBuilder).asChildOf(null as Span?)
+        inOrder(mockSpan, mockScope) {
+            verify(mockSpan).finish()
+            verify(mockScope).close()
+        }
+    }
+
+    @Test
+    fun `M create span around coroutine W runBlockingTraced() with Unconfined dispatcher`() {
+        // Given
+        var lambdaCalled = false
+        whenever(mockTracer.activeSpan()) doReturn mockParentSpan
+        whenever(mockSpanBuilder.asChildOf(mockParentSpan)) doReturn mockSpanBuilder
+
+        // When
+        runBlockingTraced(fakeOperationName, Dispatchers.Unconfined) {
+            withContext(Dispatchers.Default) {
+                lambdaCalled = true
+            }
+        }
+        Thread.sleep(100)
+
+        // Then
+        assertThat(lambdaCalled).isTrue()
+        verify(mockSpanBuilder).asChildOf(mockParentSpan)
+        verify(mockTracer, never()).activateSpan(mockSpan)
+        verify(mockSpan).finish()
+    }
+
+    @Test
+    fun `M create span around coroutine W asyncTraced()`(
+        @StringForgery data: String
+    ) {
+        // Given
+        var lambdaCalled = false
+        var result: String? = null
+        whenever(mockTracer.activeSpan()) doReturn mockParentSpan
+        whenever(mockSpanBuilder.asChildOf(mockParentSpan)) doReturn mockSpanBuilder
+
+        // When
+        runBlocking {
+            val deferred = asyncTraced(
+                fakeOperationName,
+                Dispatchers.IO
+            ) {
+                lambdaCalled = true
+                data
+            }
+            result = deferred.await()
+        }
+        Thread.sleep(100)
+
+        // Then
+        assertThat(lambdaCalled).isTrue()
+        assertThat(result).isEqualTo(data)
+        verify(mockSpanBuilder).asChildOf(mockParentSpan)
+        inOrder(mockSpan, mockScope) {
+            verify(mockSpan).finish()
+            verify(mockScope).close()
+        }
+    }
+
+    @Test
+    fun `M create span around coroutine W asyncTraced() without parents`(
```

Comment:
maybe we should have a test for when using `asyncTraced` + `awaitTraced` together ?

Reply:
Hmmm we could but I'm not sure what it would add to the existing tests, unless we also want to test the wait Kotlin's async and await is implemented.

Reply:
I am thinking more to the Span continuation as based on your implementation if the `asyncTraced` and `awaitTraced` are called in the same thread for example then the `Span` started in await should be the child of the `Span` started in `async` and if they are running in separated threads then they should be standalone `Spans`. It would be nice to have but it's your choice.

Reply:
Indeed the link can exist between those spans if they're performed on the same thread, but that's a consequence of the Span creation mechanism themselves, and not of what those extension methods do. Still I'll add those tests, but 🤷